src/osd/PG.cc

```diff
@@ -4484,6 +4484,29 @@ bool PG::may_need_replay(const OSDMapRef osdmap) const
   return crashed;
 }
 
+void PG::check_full_transition(OSDMapRef lastmap, OSDMapRef osdmap)
+{
+  if (osdmap->test_flag(CEPH_OSDMAP_FULL) &&
+      !lastmap->test_flag(CEPH_OSDMAP_FULL)) {
+    dout(10) << " cluster was marked full in " << osdmap->get_epoch()
+	     << dendl;
+    info.history.last_epoch_marked_full = osdmap->get_epoch();
```

Comment:
The only downside to the quick return is that we don't see a log message that the pool went full if both happen in the same epoch.  I'll restructure so that there is a single assignment.